### PR TITLE
Incorporating Greymass Fuel into the sign transaction process (web wallet)

### LINKED ISSUE #63: incorporate Greymass Fuel into the "Sign Transaction" process (web-wallet)
## Description
We want to incorporate Greymass Fuel services to provide seamless services for new people. One step in this direction is the incorporation of resource-free access solutions to cover the need of those who don't have and don't want to have reserved resources for their own, like new users.

## Task

Take the information listed [here](https://github.com/telosnetwork/epics/issues/18#issuecomment-1325091417) and implement a nice and user-friendly solution to incorporate Greymass Fuel services into the global signing process.

## Acceptance Criteria
- don't use Fuel for all transactions, only for those which are needed (e.g: normal try fails because of lack of resources)
- The user must be prompted to manually accept to use of this service.
- A "remember my decision" checkbox can be used to store the choice in local storage and don't ask again.
- The user should be able to turn the service off completely
- If the signing process automatically tries again using Fue, the user must receive a notification of that try.

## Note
This issue is part of the bigger issue
- https://github.com/telosnetwork/epics/issues/18

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #63

## Description
Having finished the same solution for the OBE website and approved by Aaron cox from Graymass himself, I migrate that solution to this website making minor adjustments (like migrating to javascript and integrating with i18n library for translation).

## Test scenarios
This is very difficult to test. You need to log in with an account with limited resources and then attempt to push a transaction that needs that resource.

## Checklist:
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file

### CODE REVIEW

File: src/api/fuel.js [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+  user = null;
+  constructor(user/*: User*/) {
+    super();
+    this.user = user;
+  }
+
+  async signTransaction(
+    originalTransaction/*: AnyTransaction*/,
+    originalconfig/*: SignTransactionConfig*/
+  )/*: Promise<SignTransactionResponse>*/ {
```
Comment (karynemayer) [inline src/api/fuel.js:48]: why are these commented out? 🤔

File: src/api/fuel.js [hunk 2]
```diff
@@ -0,0 +1,531 @@
+/* eslint-disable quotes */
```
Comment (karynemayer) [inline src/api/fuel.js:1]: why are we disabling the quotes here?

File: src/api/fuel.js [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+    // this are the normal texts for random wallet.
+    const cancel/*: string | boolean*/ = GreymassFuelService.globals.$t('api.reject');
+    const ok = GreymassFuelService.globals.$t('api.confirm');
+    let message = GreymassFuelService.globals.$t('api.greymass_fuel_message');
+
+    try {
+      if (typeof fees == 'string') {
+        message = GreymassFuelService.globals.$t('api.greymass_fuel_message_fees', { fees });
+      }
+    } catch (e) {}
```
Comment (karynemayer) [inline src/api/fuel.js:288]: shouldn't we at least add a `console.error` here?

**Review by poplexity — APPROVED** [review]
👍 

Looks good, please address @karynemayer's comments before merge.

File: src/api/fuel.js (hunk 1 above)
Comment (Viterbo) [reply src/api/fuel.js:48]: Yes, the idea es to enable all types once ty is enabled in this project.
@karynemayer, do you prefer them to be removed?

File: src/api/fuel.js (hunk 3 above)
Comment (Viterbo) [reply src/api/fuel.js:288]: I don't even know why the try-catch is there. I removed it.

File: src/api/fuel.js (hunk 2 above)
Comment (Viterbo) [reply src/api/fuel.js:1]: opps.. my bad!

Removed.

**Review by karynemayer — APPROVED** [review]

**Pull request merged**